### PR TITLE
Support date range in ;plot rating

### PR DESCRIPTION
**Pull request by aryanc403**

### CODE REVIEW

**Comment by aryanc403** [comment]
Fixes #313

File: tle/cogs/graphs.py [hunk 1]
```diff
@@ -205,14 +205,20 @@ async def plot(self, ctx):
         use a server member's name instead by prefixing it with '!'."""
         await ctx.send_help('plot')
 
-    @plot.command(brief='Plot Codeforces rating graph', usage='[+zoom] [handles...]')
+    @plot.command(brief='Plot Codeforces rating graph', usage='[+zoom] [handles...] [d>=[[dd]mm]yyyy] [d<[[dd]mm]yyyy]')
     async def rating(self, ctx, *args: str):
         """Plots Codeforces rating graph for the handles provided."""
+
+        filt = cf_common.SubFilter()
```
Comment (krofna) [inline tle/cogs/graphs.py:212]: Kinda dirty to reuse SubFilter. I guess if we rename it appropriately, rename filter into filter_subs and add another function for filtering rating changes then its fine.

File: tle/cogs/graphs.py [hunk 2]
```diff
@@ -36,8 +36,8 @@ def nice_sub_type(types):
                 'PRACTICE':'Practice: {}'}
     return [nice_map[t] for t in types]
 
-def _plot_rating(resp, mark='o', labels: List[str] = None):
-    labels = [''] * len(resp) if labels is None else labels
+def _plot_rating(resp, mark='o'):
+    labels = [''] * len(resp)
```
[the remark this answers is not in the document]
Comment (aryanc403) [reply tle/cogs/graphs.py:40]: This function is called at 2 places.
https://github.com/cheran-senthil/TLE/blob/master/tle/cogs/graphs.py#L225
and https://github.com/cheran-senthil/TLE/blob/master/tle/cogs/graphs.py#L435.
At both places, we don't pass labels.

[the remark this answers is not in the document]
Comment (krofna) [reply tle/cogs/graphs.py:40]: then remove labels

File: tle/cogs/graphs.py [hunk 3]
```diff
@@ -51,7 +51,7 @@ def _plot_rating(resp, mark='o', labels: List[str] = None):
                  markersize=3,
                  markerfacecolor='white',
                  markeredgewidth=0.5,
-                 label=label)
+                 label='')
```
Comment (krofna) [inline tle/cogs/graphs.py:54]: you dont have to pass this argument at all

File: tle/cogs/graphs.py [hunk 4]
```diff
@@ -205,13 +204,17 @@ async def plot(self, ctx):
...
         await ctx.send_help('plot')
 
-    @plot.command(brief='Plot Codeforces rating graph', usage='[+zoom] [handles...]')
+    @plot.command(brief='Plot Codeforces rating graph', usage='[+zoom] [handles...] [d>=[[dd]mm]yyyy] [d<[[dd]mm]yyyy]')
     async def rating(self, ctx, *args: str):
         """Plots Codeforces rating graph for the handles provided."""
+
+        filt = cf_common.SubFilter()
+        args = filt.parse(args)
         (zoom,), args = cf_common.filter_flags(args, ['+zoom'])
```
Comment (aryanc403) [inline tle/cogs/graphs.py:213]: +zoom break. SubFilter takes +zoom as problem tag.

Comment (aryanc403) [reply tle/cogs/graphs.py:213]: Fixed in #318

**Pull request merged**